stablehlo/dialect/VhloOps.td

```diff
@@ -336,86 +336,86 @@ def VHLO_AfterAllOpV1 : VHLO_Op<"after_all"> {
 }
 
 def VHLO_IfOpV1 : VHLO_Op<"if"> {
-  let arguments = (ins VHLO_PredTensor:$pred);
+  let arguments = (ins VHLO_AnyType:$pred);
   let regions = (region SizedRegion<1>:$true_branch,
                         SizedRegion<1>:$false_branch);
-  let results = (outs Variadic<VHLO_TensorOrToken>);
+  let results = (outs Variadic<VHLO_AnyType>);
 }
 
 def VHLO_CaseOpV1 : VHLO_Op<"case"> {
-  let arguments = (ins I32Tensor:$index);
+  let arguments = (ins VHLO_AnyType:$index);
   let regions = (region VariadicRegion<SizedRegion<1>>:$branches);
-  let results = (outs Variadic<VHLO_TensorOrToken>);
+  let results = (outs Variadic<VHLO_AnyType>);
 }
 
 def VHLO_WhileOpV1 : VHLO_Op<"while"> {
-  let arguments = (ins Variadic<VHLO_TensorOrToken>:$operand);
+  let arguments = (ins Variadic<VHLO_AnyType>:$operand);
   let regions = (region SizedRegion<1>:$cond, SizedRegion<1>:$body);
-  let results = (outs Variadic<VHLO_TensorOrToken>);
+  let results = (outs Variadic<VHLO_AnyType>);
 }
 
 def VHLO_AllGatherOpV1 : VHLO_Op<"all_gather", "0.3.0", "0.3.0"> {
   let arguments = (ins
-    VHLO_Tensor:$operand,
-    I64Attr:$all_gather_dim,
-    I64ElementsAttr:$replica_groups,
-    OptionalAttr<VHLO_ChannelHandle>:$channel_handle
+    VHLO_AnyType:$operand,
+    VHLO_AnyAttr:$all_gather_dim,
+    VHLO_AnyAttr:$replica_groups,
+    OptionalAttr<VHLO_AnyAttr>:$channel_handle
   );
-  let results = (outs VHLO_Tensor);
+  let results = (outs VHLO_AnyType);
 }
 
 def VHLO_AllGatherOpV2 : VHLO_Op<"all_gather_v2", "0.4.0"> {
   let arguments = (ins
-    VHLO_Tensor:$operand,
-    I64Attr:$all_gather_dim,
-    I64ElementsAttr:$replica_groups,
-    OptionalAttr<VHLO_ChannelHandle>:$channel_handle,
+    VHLO_AnyType:$operand,
+    VHLO_AnyAttr:$all_gather_dim,
+    VHLO_AnyAttr:$replica_groups,
+    OptionalAttr<VHLO_AnyAttr>:$channel_handle,
     UnitAttr:$use_global_device_ids
```

Comment:
`VHLO_AnyAttr`. (Here and for all other occurrences of `use_global_device_ids`).

Reply:
`UnitAttr` is a strange one because it is defined by omission or presence.. Removing causes tests to fail that "use_global_device_ids" is unset.

Reply:
`OptionalAttr<VHLO_AnyAttr>` perhaps?

Reply:
I'm a bit surprised that worked but it worked. `UnitAttr` is not covered by `AnyAttr`, but is covered by `OptionalAttr<AnyAttr>`. I suppose that fits its definition. Resolved